packages/nx/src/utils/params.ts

```diff
@@ -555,20 +553,33 @@ function setPropertyDefault(
         setDefaultsInObject(valueInArray, items.properties || {}, definitions)
       );
     } else if (!opts[propName] && schema.default) {
-      opts[propName] = schema.default;
+      defaultValueToSet = schema.default;
     }
   } else {
-    const wasUndefined = opts[propName] === undefined;
-    if (wasUndefined) {
-      // We need an object to set values onto
-      opts[propName] = {};
+    if (opts[propName] === undefined && schema.default !== undefined) {
+      defaultValueToSet = schema.default;
     }
 
-    setDefaultsInObject(opts[propName], schema.properties || {}, definitions);
+    if (schema.type === 'object') {
+      const wasUndefined = opts[propName] === undefined;
+      if (!wasUndefined) {
+        setDefaultsInObject(
+          opts[propName],
+          schema.properties || {},
+          definitions
+        );
+      }
+    }
+  }
 
-    // If the property was initially undefined but no properties were added, we remove it again instead of having an {}
-    if (wasUndefined && Object.keys(opts[propName]).length === 0) {
-      delete opts[propName];
+  if (defaultValueToSet !== undefined) {
+    try {
+      validateProperty(propName, defaultValueToSet, schema, definitions);
+      opts[propName] = defaultValueToSet;
+    } catch (e) {
+      // If the default value is invalid, we don't set it...
```

Comment:
future: add a warning saying that the property was not applied because it was invalid and why it was invalid.